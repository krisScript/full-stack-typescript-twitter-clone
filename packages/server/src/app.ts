import express, { Request, Response, NextFunction, Application } from 'express';
import bodyParser from 'body-parser';
import helmet from 'helmet';
import multer from 'multer';
import compression from 'compression';
import userRoutes from '@routes/userRoutes';
import tweetRoutes from '@routes/tweetRoutes';
import { CustomError } from '@utilities/CustomError';
import fileFilter from '@customMiddleware/fileFilter';

const app: Application = express();

app.use(helmet());
app.use(compression());
app.use(bodyParser.json());

app.use((req: Request, res: Response, next: NextFunction): void => {
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader(
    'Access-Control-Allow-Methods',
    'OPTIONS, GET, POST, PUT, PATCH, DELETE',
  );
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');
  next();
});
const fileStorage = multer.diskStorage({
  // @ts-ignore
  destination: (req: Request, file, cb: (...args: any) => any): void => {
    cb(null, './images');
  },
  filename: (req: Request, file, cb: (...args: any) => any): void => {
    cb(
      null,
      `${new Date().toISOString().replace(/:/g, '-')}-${file.originalname}`,
    );
  },
});

app.use(multer({ storage: fileStorage, fileFilter }).single('image'));
app.use('/images', express.static('./images'));

app.use(userRoutes);
app.use(tweetRoutes);

app.use(
  (
    error: CustomError,
    req: Request,
    res: Response,
    next: NextFunction,
  ): void => {
    console.log(error);
    const status = error.status || 500;
    const { message } = error;
    const response = error.data ? { data: error.data, message } : { message };
    res.status(status).json(response);
  },
);

export default app;
