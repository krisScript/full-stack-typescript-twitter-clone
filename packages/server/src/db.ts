import mongoose from "mongoose";

const mongoURI: string = `mongodb+srv://${process.env.MONGO_USER}:${process.env.MONGO_PASSWORD}@cluster0-zmcyw.mongodb.net/${process.env.MONGO_DATABASE}?retryWrites=true`;
const db = async (): Promise<void> => {
  try {
    await mongoose.connect(mongoURI, { useNewUrlParser: true });
  } catch (err) {}
};

export default db;
