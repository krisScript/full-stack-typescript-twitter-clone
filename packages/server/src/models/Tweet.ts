import { NextFunction } from 'express';
import mongoose, { Schema } from 'mongoose';
import Tweet from '@customTypes/Tweet';

const TweetSchema: Schema = new Schema({
  type: { type: String, required: true, enum: ['text', 'image', 'retweet'] },
  text: { type: String, minlength: 1, maxlength: 100 },
  user: {
    type: Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  retweets: { type: Number, default: 0 },
  likes: { type: Number, default: 0 },
  comments: { type: Number, default: 0 },
  date: {
    type: Date,
    default: Date.now,
  },
});

TweetSchema.pre('find', function(next: NextFunction): void {
  this.populate([
    { path: 'user', select: 'username' },
    { path: 'community', select: 'name description subscribers theme' },
  ]);
  next();
});
export default mongoose.model<Tweet>('Tweet', TweetSchema);
