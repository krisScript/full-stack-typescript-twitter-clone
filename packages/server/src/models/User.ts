import mongoose, { Schema } from 'mongoose';
import User from '@customTypes/User';

const UserSchema: Schema = new Schema({
  username: {
    required: true,
    type: String,
    minlength: 1,
    maxlength: 50,
    unique: true,
  },
  handle: {
    required: true,
    type: String,
    minlength: 1,
    maxlength: 50,
    unique: true,
  },
  email: { required: true, type: String, minlength: 3, unique: true },
  password: { required: true, type: String, minlength: 12 },
  website: { required: false, type: String, minlength: 3 },
  confirmed: { type: Boolean, default: false },
  profilePhoto: { type: String },
  headerPhoto: { type: String },
  date: {
    type: Date,
    default: Date.now,
  },
  following: [
    {
      type: Schema.Types.ObjectId,
      ref: 'User',
    },
  ],
  followers: {
    type: Number,
    default: 0,
  },
  likes: [
    {
      source: {
        type: mongoose.Types.ObjectId,
        required: true,
        refPath: 'likes.ref',
      },
      ref: {
        type: String,
        required: true,
        enum: ['Tweet', 'Reply'],
      },
    },
  ],
  bookmarks: [
    {
      source: {
        type: mongoose.Types.ObjectId,
        required: true,
        refPath: 'bookmarks.ref',
      },
      ref: {
        type: String,
        required: true,
        enum: ['Tweet', 'Reply'],
      },
    },
  ],
});

export default mongoose.model<User>('User', UserSchema);
