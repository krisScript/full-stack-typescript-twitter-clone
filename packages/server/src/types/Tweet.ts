import CommonTweet from '@twtr/common/source/types/Tweet';
import { Document } from 'mongoose';

export default interface Tweet extends CommonTweet, Document {
  user: string;
  retweet: string;
  reply: string;
}
