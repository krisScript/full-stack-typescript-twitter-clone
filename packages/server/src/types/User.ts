import CommonUser from '@twtr/common/source/types/User';
import { Schema, Document } from 'mongoose';

export default interface User extends CommonUser, Document {
  password: string;
  confirmed: boolean;
  following: Schema.Types.ObjectId[];
  likes: Schema.Types.ObjectId[];
  bookmarks: Schema.Types.ObjectId[];
}
