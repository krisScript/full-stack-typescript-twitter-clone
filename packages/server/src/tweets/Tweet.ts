import mongoose, { Schema, Query } from 'mongoose';
import Tweet from '../types/Tweet';

export const TweetSchema: Schema = new Schema({
  type: {
    type: String,
    required: true,
    enum: ['text', 'link', 'retweet', 'reply'],
  },
  text: { type: String, maxlength: 500 },
  image: { type: String },
  link: { type: String },
  user: {
    type: Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  retweet: {
    type: Schema.Types.ObjectId,
    ref: 'Tweet',
  },
  reply: { type: Schema.Types.ObjectId, ref: 'Tweet' },
  retweets: { type: Number, default: 0 },
  likes: { type: Number, default: 0 },
  replies: { type: Number, default: 0 },
  date: {
    type: Date,
    default: Date.now,
  },
});
//@ts-ignore
async function populateFields(this: Query<any>): Promise<void> {
  this.populate([
    { path: 'user', select: 'username handle avatar' },
    { path: 'reply', select: 'user' },
    { path: 'retweet' },
  ]);
}

TweetSchema.pre('find', populateFields);

TweetSchema.pre('populate', populateFields);

export default mongoose.model('Tweet', TweetSchema);
