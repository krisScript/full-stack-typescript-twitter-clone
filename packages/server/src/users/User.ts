import mongoose, { Schema } from 'mongoose';
import User from '../types/User';
import duplicationErrorHandler from '../middleware/duplicationErrorHandler';
//@ts-ignore
import uniqueValidator from 'mongoose-unique-validator';

const UserSchema: Schema = new Schema({
  username: {
    required: true,
    type: String,
    minlength: 1,
    maxlength: 50,
    unique: true,
  },
  handle: {
    required: true,
    type: String,
    minlength: 1,
    maxlength: 50,
    unique: true,
  },
  email: { required: true, type: String, minlength: 3, unique: true },
  password: { required: true, type: String, minlength: 12 },
  isConfirmed: { type: Boolean, default: true },
  avatar: {
    type: String,
  },
  cover: { type: String },
  date: {
    type: Date,
    default: Date.now,
  },
  following: [
    {
      type: Schema.Types.ObjectId,
      ref: 'User',
    },
  ],
  followers: {
    type: Number,
    default: 0,
  },
  likes: [
    {
      type: Schema.Types.ObjectId,
      ref: 'Tweet',
    },
  ],
  retweets: [
    {
      type: Schema.Types.ObjectId,
      ref: 'Tweet',
    },
  ],
  replies: [
    {
      type: Schema.Types.ObjectId,
      ref: 'Tweet',
    },
  ],

  bookmarks: [
    {
      type: Schema.Types.ObjectId,
      ref: 'Tweet',
    },
  ],
});

UserSchema.index({ handle: 'text' });

UserSchema.plugin(uniqueValidator);

export default mongoose.model<User>('User', UserSchema);
