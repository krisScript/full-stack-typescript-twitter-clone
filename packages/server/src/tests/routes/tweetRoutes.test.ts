import mongoose from 'mongoose';
import request from 'supertest';
import jwt from 'jsonwebtoken';
import mockFs from 'mock-fs';
import mjml from 'mjml';
import app from 'src/app';
import User from '@models/User';
import db from 'src/db';
import Tweet from '@models/Tweet';

const port = process.env.PORT || 8080;
const mockTemplate = 'MockTemplate';
(mjml as jest.Mock).mockReturnValue(mockTemplate);
jest.mock('mjml');

describe('tweetRoutes', (): void => {
  beforeAll(
    async (): Promise<void> => {
      await mongoose.disconnect();
      await db();
      app.listen(port);
      await Tweet.deleteMany({}).exec();
      await User.deleteMany({}).exec();
    },
  );
  afterEach(
    async (): Promise<void> => {
      mockFs.restore();
      await Tweet.deleteMany({}).exec();
      await User.deleteMany({}).exec();
    },
  );
  afterAll(
    async (): Promise<void> => {
      await mongoose.disconnect();
    },
  );
  const username = 'username';
  const handle = 'testUserHandle';
  const email = 'testmail@mail.com';
  const password = 'testPassword';
  const text =
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique vel alias, amet corporis modi corrupti.';
  const link = 'https://fakeLink.fakeLink';
  const secret = process.env.SECRET;
  describe('post /tweets', (): void => {
    it('should create a new text tweet', async (): Promise<void> => {
      expect.assertions(1);
      const newUser = new User({
        username,
        handle,
        email,
        password,
      });
      await newUser.save();
      const userId = newUser._id;

      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const type = 'text';
      const response = await request(app)
        .post('/tweets')
        .send({
          type,
          text,
        })
        .set('Authorization', `Bearer ${token}`);
      expect(response.status).toBe(200);
    });
    it('should create a new link tweet', async (): Promise<void> => {
      expect.assertions(1);
      const newUser = new User({
        username,
        handle,
        email,
        password,
      });
      await newUser.save();
      const userId = newUser._id;

      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const type = 'link';
      const response = await request(app)
        .post('/tweets')
        .send({
          type,
          linkUrl: link,
        })
        .set('Authorization', `Bearer ${token}`);
      expect(response.status).toBe(200);
    });
    it('should create a new retweet', async (): Promise<void> => {
      expect.assertions(1);
      const newUser = new User({
        username,
        handle,
        email,
        password,
      });
      await newUser.save();
      const userId = newUser._id;
      const tweet = new Tweet({
        type: 'link',
        link,
        user: userId,
      });
      await tweet.save();
      const retweetId = tweet._id;
      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const type = 'retweet';
      const response = await request(app)
        .post('/tweets')
        .send({
          type,
          text,
          retweetId,
        })
        .set('Authorization', `Bearer ${token}`);
      expect(response.status).toBe(200);
    });
    it('should create a new reply tweet', async (): Promise<void> => {
      expect.assertions(1);
      const newUser = new User({
        username,
        handle,
        email,
        password,
      });
      await newUser.save();
      const userId = newUser._id;
      const tweet = new Tweet({
        type: 'link',
        link,
        user: userId,
      });
      await tweet.save();
      const replyId = tweet._id;
      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const type = 'reply';
      const response = await request(app)
        .post('/tweets')
        .send({
          type,
          text,
          replyId,
        })
        .set('Authorization', `Bearer ${token}`);
      expect(response.status).toBe(200);
    });
    it('should create a new image tweet', async (): Promise<void> => {
      expect.assertions(1);
      const newUser = new User({
        username,
        handle,
        email,
        password,
      });
      await newUser.save();
      const userId = newUser._id;
      mockFs({
        './images': {
          'test.jpg': Buffer.from([8, 6, 7, 5, 3, 0, 9]),
        },
      });
      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const type = 'text';
      const response = await request(app)
        .post('/tweets')
        .field({
          type,
          text,
        })
        .attach('image', './images/test.jpg')
        .set('Authorization', `Bearer ${token}`);
      expect(response.status).toBe(200);
    });

    it("should throw an error with a status of 400: BadRequest when the tweet type and the content don't match", async (): Promise<
      void
    > => {
      expect.assertions(1);
      const newUser = new User({
        username,
        handle,
        email,
        password,
      });
      await newUser.save();
      const userId = newUser._id;

      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const type = 'link';
      const response = await request(app)
        .post('/tweets')
        .send({
          type,
          text,
        })
        .set('Authorization', `Bearer ${token}`);
      expect(response.status).toBe(400);
    });
    it('should throw an error with a status of 401: Unauthorized when there is no authorization header or its contents are invalid', async (): Promise<
      void
    > => {
      expect.assertions(1);
      const type = 'text';
      const response = await request(app)
        .post('/tweets')
        .send({
          type,
          text,
        });
      expect(response.status).toBe(401);
    });
  });
  describe('patch /tweets/:tweetId', (): void => {
    it('should update a text tweet', async (): Promise<void> => {
      expect.assertions(2);
      const newText = 'newTestText';
      const userId = mongoose.Types.ObjectId().toString();
      const newTweet = new Tweet({
        type: 'text',
        text,
        user: userId,
      });
      await newTweet.save();
      const tweetId = newTweet._id;
      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const response = await request(app)
        .patch(`/tweets/${tweetId}`)
        .set('Authorization', `Bearer ${token}`)
        .send({
          type: 'text',
          text: newText,
        });
      const tweet = await Tweet.findById(tweetId);
      if (!tweet) {
        return;
      }
      expect(response.status).toBe(204);
      expect(tweet.text).toMatch(newText);
    });
    it('should update a link tweet', async (): Promise<void> => {
      expect.assertions(2);
      const newLink = 'https://fakeNewLink.com';
      const userId = mongoose.Types.ObjectId().toString();
      const newTweet = new Tweet({
        type: 'link',
        link,
        user: userId,
      });
      await newTweet.save();
      const tweetId = newTweet._id;
      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const response = await request(app)
        .patch(`/tweets/${tweetId}`)
        .set('Authorization', `Bearer ${token}`)
        .send({
          type: 'link',
          linkUrl: newLink,
        });
      const tweet = await Tweet.findById(tweetId);
      if (!tweet) {
        return;
      }
      expect(response.status).toBe(204);
      expect(tweet.link).toMatch(newLink);
    });
    it("should throw an error with a status of 400: BadRequest when the req body doesn't pass validation", async (): Promise<
      void
    > => {
      expect.assertions(1);
      const userId = mongoose.Types.ObjectId().toString();
      const newTweet = new Tweet({
        type: 'link',
        link,
        user: userId,
      });
      await newTweet.save();
      const tweetId = newTweet._id;
      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const response = await request(app)
        .patch(`/tweets/${tweetId}`)
        .set('Authorization', `Bearer ${token}`);
      expect(response.status).toBe(400);
    });
    it('should throw an error with a status of 401: Unauthorized when there is no authorization header or its contents are invalid', async (): Promise<
      void
    > => {
      expect.assertions(1);
      const newLink = 'https://fakeNewLink.com';
      const userId = mongoose.Types.ObjectId().toString();
      const newTweet = new Tweet({
        type: 'link',
        link,
        user: userId,
      });
      await newTweet.save();
      const tweetId = newTweet._id;
      const response = await request(app)
        .patch(`/tweets/${tweetId}`)
        .send({
          type: 'link',
          linkUrl: newLink,
        });
      expect(response.status).toBe(401);
    });
    it('should throw an error with a status of 404: NotFound when the tweet is not found', async (): Promise<
      void
    > => {
      const newLink = 'https://fakeNewLink.com';
      expect.assertions(1);
      const userId = mongoose.Types.ObjectId().toString();
      const tweetId = mongoose.Types.ObjectId();
      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const response = await request(app)
        .patch(`/tweets/${tweetId}`)
        .set('Authorization', `Bearer ${token}`)
        .send({
          type: 'link',
          linkUrl: newLink,
        });
      expect(response.status).toBe(404);
    });
  });
  describe('delete /tweets/:tweetId', (): void => {
    it('should delete a tweet', async (): Promise<void> => {
      expect.assertions(2);
      const userId = mongoose.Types.ObjectId().toString();
      const newTweet = new Tweet({
        type: 'text',
        text,
        user: userId,
      });
      await newTweet.save();
      const tweetId = newTweet._id;
      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const response = await request(app)
        .delete(`/tweets/${tweetId}`)
        .set('Authorization', `Bearer ${token}`);
      const tweet = await Tweet.findById(tweetId);
      expect(response.status).toBe(204);
      expect(tweet).toBeNull();
    });
    it('should throw an error with a status of 401: Unauthorized when there is no authorization header or its contents are invalid', async (): Promise<
      void
    > => {
      expect.assertions(2);
      const userId = mongoose.Types.ObjectId().toString();
      const unauthorizedUserId = mongoose.Types.ObjectId().toString();
      const newTweet = new Tweet({
        type: 'text',
        text,
        user: userId,
      });
      await newTweet.save();
      const tweetId = newTweet._id;
      const token = jwt.sign(
        {
          userId: unauthorizedUserId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const response = await request(app)
        .delete(`/tweets/${tweetId}`)
        .set('Authorization', `Bearer ${token}`);
      const tweet = await Tweet.findById(tweetId);
      expect(response.status).toBe(401);
      expect(tweet).not.toBeNull();
    });
    it('should throw an error with a status of 404: NotFound when the tweet is not found', async (): Promise<
      void
    > => {
      expect.assertions(1);
      const userId = mongoose.Types.ObjectId().toString();
      const tweetId = mongoose.Types.ObjectId();
      const token = jwt.sign(
        {
          userId,
        },
        secret,
        { expiresIn: '1h' },
      );
      const response = await request(app)
        .delete(`/tweets/${tweetId}`)
        .set('Authorization', `Bearer ${token}`);
      expect(response.status).toBe(404);
    });
  });
  describe('get /tweets/:tweetId', (): void => {
    it('should get a tweet', async (): Promise<void> => {
      expect.assertions(2);
      const user = new User({
        username,
        handle,
        email,
        password,
      });
      await user.save();
      const userId = user._id;
      const newTweet = new Tweet({
        type: 'text',
        text,
        user: userId,
      });
      await newTweet.save();
      const tweetId = newTweet._id;
      const response = await request(app).get(`/tweets/${tweetId}`);
      const { tweet } = response.body.data;
      expect(response.status).toBe(200);
      expect(tweet._id.toString()).toMatch(tweetId.toString());
    });
    it('should throw an error with a status of 404: NotFound when the tweet is not found', async (): Promise<
      void
    > => {
      expect.assertions(1);
      const tweetId = mongoose.Types.ObjectId();
      const response = await request(app).get(`/tweets/${tweetId}`);
      expect(response.status).toBe(404);
    });
  });
  describe('get /users/:userId/tweets', (): void => {
    it('should get a list of tweets', async (): Promise<void> => {
      expect.assertions(1);
      const userId = mongoose.Types.ObjectId().toString();
      const newTweet = new Tweet({
        type: 'text',
        text,
        user: userId,
      });
      await newTweet.save();
      const response = await request(app).get(`/users/${userId}/tweets`);
      expect(response.status).toBe(200);
    });
  });
  describe('get /tweets', (): void => {
    it('should get a list of tweets', async (): Promise<void> => {
      expect.assertions(1);
      const userId = mongoose.Types.ObjectId().toString();
      const newTweet = new Tweet({
        type: 'text',
        text,
        user: userId,
      });
      await newTweet.save();
      const response = await request(app).get(`/tweets`);
      expect(response.status).toBe(200);
    });
  });
  describe('get /tweets/:tweetId/replies', (): void => {
    it('should get a list of replies by tweet', async (): Promise<void> => {
      expect.assertions(1);
      const userId = mongoose.Types.ObjectId().toString();
      const tweetId = mongoose.Types.ObjectId().toString();
      const newReply = new Tweet({
        text,
        type: 'reply',
        user: userId,
        reply: tweetId,
      });
      await newReply.save();
      const response = await request(app).get(`/tweets/${tweetId}/replies`);
      expect(response.status).toBe(200);
    });
  });
  describe('get /users/:userId/replies', (): void => {
    it('should get a list of replies by user', async (): Promise<void> => {
      expect.assertions(1);
      const userId = mongoose.Types.ObjectId().toString();
      const tweetId = mongoose.Types.ObjectId().toString();
      const newReply = new Tweet({
        text,
        type: 'reply',
        user: userId,
        reply: tweetId,
      });
      await newReply.save();
      const response = await request(app).get(`/users/${userId}/replies`);
      expect(response.status).toBe(200);
    });
  });
});
