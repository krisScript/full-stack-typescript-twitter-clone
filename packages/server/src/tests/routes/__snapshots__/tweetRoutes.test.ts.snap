// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tweetRoutes get /tweets should get a list of tweets 1`] = `
Array [
  Object {
    "__v": 0,
    "_id": "5d597dbcd9a2fa3e40cbb3f9",
    "comments": 0,
    "date": "2019-08-18T16:33:00.381Z",
    "likes": 0,
    "retweets": 0,
    "text": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique vel alias, amet corporis modi corrupti.",
    "type": "text",
    "user": null,
  },
]
`;

exports[`tweetRoutes get /tweets should get a list of tweets 2`] = `
Object {
  "next": null,
  "prev": null,
}
`;

exports[`tweetRoutes get /users/:userId/tweets should get a list of tweets 1`] = `
Array [
  Object {
    "__v": 0,
    "_id": "5d597dbbd9a2fa3e40cbb3f7",
    "comments": 0,
    "date": "2019-08-18T16:32:59.608Z",
    "likes": 0,
    "retweets": 0,
    "text": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique vel alias, amet corporis modi corrupti.",
    "type": "text",
    "user": null,
  },
]
`;

exports[`tweetRoutes get /users/:userId/tweets should get a list of tweets 2`] = `
Object {
  "next": null,
  "prev": null,
}
`;
